src/shared/actions/transfers.js

```diff
@@ -1022,3 +1025,46 @@ export const retryFailedTransaction = (accountName, bundleHash, seedStore, quoru
             }
         });
 };
+
+/**
+ * Verifies CDA content
+ *
+ * @method verifyCDAContent
+ *
+ * @param {object} CDA data
+ *
+ * @returns {function} dispatch
+ */
+export const verifyCDAContent = (data) => {
+    return (dispatch) => {
+      try {
+          verifyCDA(Date.now() / 1000, data);
+          dispatch(setCDAContent(data));
+          dispatch(setSendAddressField(data.address));
+          if (data.expectedAmount) {
+              dispatch(setSendAmountField(data.expectedAmount.toString()));
+          }
+          if (data.message) {
+              dispatch(setSendMessageField(data.message));
+          }
+      } catch (err) {
+          if (err.message === 'Expired timeout.') {
```

Comment:
`Expired timeout.` should probably go to https://github.com/iotaledger/trinity-wallet/blob/feat/exchanges/src/shared/libs/errors.js